src/amqp.rs

```diff
@@ -0,0 +1,68 @@
+use lapin::tcp::{OwnedIdentity, OwnedTLSConfig};
+use vector_config::configurable_component;
+
+/// Connection options for AMQP.
+#[configurable_component]
+#[derive(Clone, Debug)]
+pub(crate) struct AMQPConfig {
```

Comment:
Pedantic nit, should this be `AmqpConfig`

> In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn.

Reply:
Yeah, makes sense.

Reply:
I don't think we really follow strictly in the project, I had just looked it up today and had it on my mind 😄 no demand from me to rename it

Reply:
It had to be `Amqp` in the Sources enum https://github.com/vectordotdev/vector/blob/c1fd0effead5d2de6f2217d87abd5554dc8779b4/src/sources/mod.rs#L108 in order to serialize properly. It's good to keep it consistent throughout.

Reply:
I figure we can do a normalization pass whenever we get around to finalizing our style guide 😄